src/internal/connector/onedrive/permission.go

```diff
@@ -127,6 +128,25 @@ func createRestoreFoldersWithPermissions(
 	return id, err
 }
 
+// isSame checks equality of two UserPermission objects
+func isSame(p1, p2 UserPermission) bool {
+	if p1.EntityID != "" && p1.EntityID != p2.EntityID {
```

Comment:
do we know the conditions under which the entityID or the email will be empty? Can we get comments on how that could happen please

Also, do we need to worry about checking equality across backup versions? For example, would it be possible for the child to have a struct from an older backup that used Email and the parent to be from a newer backup that used EntityID?

Reply:
I've added comments explaining these things in the code in https://github.com/alcionai/corso/pull/3117/commits/7e1ae0bc68153cbbb03c4e95948f77e1fe4d692f